src/protocols/secure_channel/CASESession.cpp

```diff
@@ -695,16 +681,20 @@ CHIP_ERROR CASESession::HandleSigmaR2(System::PacketBufferHandle & msg)
 
     SuccessOrExit(err = decryptedDataTlvReader.Next());
     VerifyOrExit(TLV::TagNumFromTag(decryptedDataTlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
-    responderOpCertLen = static_cast<uint16_t>(decryptedDataTlvReader.GetLength());
-    SuccessOrExit(err = decryptedDataTlvReader.GetBytes(responderOpCert, responderOpCertLen));
+
+    responderOpCertLen = static_cast<size_t>(decryptedDataTlvReader.GetLength());
+    // We use `sizeof(responderOpCert)` rather than `responderOpCertLen` since GetBytes()
+    // validates that the destination buffer is large enough for the equivalent of GetLength().
+    // If we used untrusted `responderOpCertLen` directly, and a bad value was provided,
+    // it could overrun stack without being caught.
```

Comment:
We should (followup, obviously) think about a better/safer "get the bytes" API on TLVReader, which takes a "buffer + size" and outputs "bytes and length".  Most simply, taking `MutableByteSpan &` or something.

Reply:
Opened issue